bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,121 @@
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+
+from discord import Color, Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+
+TOP_WORDS_FILE_PATH = Path("bot/resources/fun/top_1000_used_words.txt").resolve()
+# defining a list of images that will be used for the game to represent the hangman person
+IMAGES = [
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472355056295946/hangman0.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472756744814613/hangman1.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472808699629578/hangman2.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472862441267230/hangman3.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472950991396864/hangman4.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883472999431430204/hangman5.png",
+    "https://cdn.discordapp.com/attachments/859123972884922418/883473051277226015/hangman6.png"
+]
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with TOP_WORDS_FILE_PATH.open(mode="r", encoding="utf-8") as file:
+            self.all_words = [line.strip('\n') for line in file.readlines()]
+
+    @commands.command()
+    async def hangman(
+            self,
+            ctx: commands.Context,
+            min_length: str = "0",
+            max_length: str = "25",
+            min_unique_letters: str = "0",
+            max_unique_letters: str = "25"
+    ) -> None:
+        """
+        Play against the bot, where you have to guess the word it has provided!
+
+        You have 7 tries, and you have to guess the letters that are in the word, or you lose!
+        """
+        # filtering the list of all words depending on the configuration
+        filtered_words = [word for word in self.all_words
+                          if int(min_length) < len(word) < int(max_length)
+                          and int(min_unique_letters) < len(set(word)) < int(max_unique_letters)]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title="With the filters you provided, I could not find any words that fit all the filters specified..",
```

Comment:
I believe you used literal concatenation earlier, why not continue it? This is a little long. Maybe consider using:
```py
title=(
    "With the filters you provided, "
    "I could not find any words that fit all the filters specified."
)
```